tasks/files/snapshot.py

```diff
@@ -645,90 +1048,209 @@ def validate_args(args):
     return True
 
 
+def get_required_space(required_space_str):
+    required_space = 0
+
+    if required_space_str:
+        try:
+            required_space = int(required_space_str)
+        except ValueError:
+            print("-r  --required_space: conversion error", args.operation)
+            sys.exit(SnapshotStatus.ERROR_CMD_INVALID)
+
+    return required_space
+
+
+def print_result(rc, message):
+    if rc != SnapshotStatus.SNAPSHOT_OK:
+        print(message, file=sys.stderr)
+        logger.info("exit code: %d: %s", rc, message)
+
+
+def validate_name_volume(snapset_json):
+    try:
+        snapset_name = snapset_json["name"]
+    except IndexError:
+        return (SnapshotStatus.ERROR_JSON_PARSER_ERROR,
+                "snapset must include a name entry")
+
+    try:
+        volume_list = snapset_json["volumes"]
+    except IndexError:
+        return (SnapshotStatus.ERROR_JSON_PARSER_ERROR,
+                "snapset must include a volumes list entry")
+
+    try:
+        for list_item in volume_list:
+            vg = list_item["vg"]
+    except IndexError:
+        return (SnapshotStatus.ERROR_JSON_PARSER_ERROR,
+                "snapset volume must include a vg entry:" + snapset_name)
+
+    try:
+        for list_item in volume_list:
+            vg = list_item["lv"]
+    except IndexError:
+        return (SnapshotStatus.ERROR_JSON_PARSER_ERROR,
+                "snapset volume must include a lv entry:" + snapset_name + " " + vg)
+
+    return SnapshotStatus.SNAPSHOT_OK, ""
+
+
+def validate_snapset_json_snapshot_cmd(snapset_json):
+    rc, message = validate_name_volume(snapset_json)
+
+    if rc != SnapshotStatus.SNAPSHOT_OK:
```

Comment:
```suggestion
```
Can the function just always return `rc, message` and have the caller check if `rc != SnapshotStatus.SNAPSHOT_OK`?

Reply:
Right... I reworked this to also check for percent_space_required when it is/is not required.